waspc/packages/wasp-config/__tests__/testFixtures.ts

```diff
@@ -71,731 +145,844 @@ type PageType = (typeof PAGE_TYPES)[number];
+  scope: "full",
+): FullNamedConfig<TsAppSpec.CrudConfig>;
+export function getCrudConfig(
+  scope: ConfigType,
+): NamedConfig<TsAppSpec.CrudConfig>;
+export function getCrudConfig(
+  scope: ConfigType,
+): NamedConfig<TsAppSpec.CrudConfig> {
   if (scope === "minimal") {
     return {
       name: "MinimalCrud",
       config: {
         entity: getEntity("task"),
         operations: getCrudOperations(scope),
       },
-    } satisfies MinimalNamedConfig<UserApi.Crud>;
+    } satisfies MinimalNamedConfig<TsAppSpec.CrudConfig>;
+  } else {
+    return {
+      name: "FullCrud",
+      config: {
+        entity: getEntity("task"),
+        operations: getCrudOperations(scope),
+      },
+    } satisfies FullNamedConfig<TsAppSpec.CrudConfig>;
   }
-
-  return {
-    name: "FullCrud",
-    config: {
-      entity: getEntity("task"),
-      operations: getCrudOperations(scope),
-    },
-  } satisfies FullNamedConfig<UserApi.Crud>;
 }
 
 export function getCrudOperations(
   scope: "minimal",
-): MinimalConfig<UserApi.CrudOperations>;
+): MinimalConfig<TsAppSpec.CrudOperations>;
 export function getCrudOperations(
   scope: "full",
-): FullConfig<UserApi.CrudOperations>;
+): FullConfig<TsAppSpec.CrudOperations>;
+export function getCrudOperations(
+  scope: ConfigType,
+): Config<TsAppSpec.CrudOperations>;
 export function getCrudOperations(
   scope: ConfigType,
-): Config<UserApi.CrudOperations> {
+): Config<TsAppSpec.CrudOperations> {
   if (scope === "minimal") {
-    return {} satisfies MinimalConfig<UserApi.CrudOperations>;
+    return {} satisfies MinimalConfig<TsAppSpec.CrudOperations>;
+  } else {
+    return {
+      get: getCrudOperationOptions(scope),
+      getAll: getCrudOperationOptions(scope),
+      create: getCrudOperationOptions(scope),
+      update: getCrudOperationOptions(scope),
+      delete: getCrudOperationOptions(scope),
+    } satisfies FullConfig<TsAppSpec.CrudOperations>;
   }
-
-  return {
-    get: getCrudOperationOptions(scope),
-    getAll: getCrudOperationOptions(scope),
-    create: getCrudOperationOptions(scope),
-    update: getCrudOperationOptions(scope),
-    delete: getCrudOperationOptions(scope),
-  } satisfies FullConfig<UserApi.CrudOperations>;
 }
 
 export function getCrudOperationOptions(
   scope: "minimal",
-): MinimalConfig<UserApi.CrudOperationOptions>;
+): MinimalConfig<TsAppSpec.CrudOperationOptions>;
 export function getCrudOperationOptions(
   scope: "full",
-): FullConfig<UserApi.CrudOperationOptions>;
+): FullConfig<TsAppSpec.CrudOperationOptions>;
 export function getCrudOperationOptions(
   scope: ConfigType,
-): Config<UserApi.CrudOperationOptions> {
+): Config<TsAppSpec.CrudOperationOptions>;
+export function getCrudOperationOptions(
+  scope: ConfigType,
+): Config<TsAppSpec.CrudOperationOptions> {
   if (scope === "minimal") {
-    return {} satisfies MinimalConfig<UserApi.CrudOperationOptions>;
+    return {} satisfies MinimalConfig<TsAppSpec.CrudOperationOptions>;
```

Comment:
Seems like this type is now broken:
```ts
// TS doesn't complain here:
return { test: 'a', setupFn: 'asdf'  } satisfies MinimalConfig<TsAppSpec.ServerConfig>;
```
And truly, if you create an alias and hover:

![image](https://github.com/user-attachments/assets/716efbe6-33aa-4427-8f82-74ba010be701)

To make matters worse, this propagates from child to parent objects:

![image](https://github.com/user-attachments/assets/f6fc19be-5fca-4178-b4ab-6bf0375feb91)

Since these config helper types have turned out to be pretty unwieldy, we should write tests for them and clearly documented how they're supposed to work.

Reply:
One small improvement I see is that we don't need the `infer` for the brand, we can just use `T` directly I believe:
```ts
type MinimalConfig<T> =
  T extends Branded<unknown, unknown>
    ? T 
    : ...
```
Fewer repetitions and less room for errors.